frontend/src/components/common/form/ParentSelect.tsx

```diff
@@ -136,28 +137,30 @@ export const ParentSelect: React.FC<IParentSelect> = ({
             }),
             x => x.parentId,
           );
-          const items = Object.keys(parents)
+          const items = Object.keys(resultGroup)
             .sort()
-            .map(parent => (
-              <Fragment key={parent}>
-                {!!results.find((x: SelectOption) => x.value === parent) && (
+            .map(parentId => (
+              <Fragment key={parentId}>
+                {!!results.find((x: SelectOption) => x.parentId === +parentId) && (
                   <Menu.Header
                     onClick={() =>
                       enableMultiple
                         ? handleMultiSelectHeaderClick(
-                            results.filter(x => x.parentId?.toString() === parent),
+                            results.filter(x => x.parentId?.toString() === parentId),
                           )
-                        : handleMenuHeaderClick(results.find(x => x.value === parent)!)
+                        : handleMenuHeaderClick(results.find(x => x.value === parentId)!)
                     }
                   >
                     <b style={{ cursor: 'pointer' }}>
+                      {/* project statuses has a different way of selecting parent value*/}
                       {field === 'statusId'
-                        ? results.find(x => x.parentId?.toString() === parent)?.parent
-                        : results.find(x => x.value === parent)?.label}
+                        ? results.find(x => x.parentId?.toString() === parentId)?.parent
+                        : results.find(x => x.value === parentId)?.label}
```

Comment:
so x.value is a string but x.parentId isn't and so we need the toString()?

Reply:
There's something strange with out typing somewhere, going to look into it more after I finish off my last story

Reply:
But yes correct